### PR TITLE
OPSEXP-2554 Enable using activemq password with $ for transformers role

### PR DESCRIPTION
**Pull request by pmacius**
Ref: OPSEXP-2554 
https://0pointer.de/public/systemd-man/systemd.exec.html#EnvironmentFile=

### CODE REVIEW

File: roles/transformers/templates/alfresco-tengine-aio.service.j2 [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+-DIMAGEMAGICK_CODERS=" + im_cfg() + "/modules-Q16HDRI/coders" %}
+{% if acs.edition == 'Enterprise' %}
+{% set java_opts = java_opts + " \
+-DACTIVEMQ_URL=failover:(" + activemq_transport + "://" + activemq_host + ":" + ports_cfg.activemq[activemq_protocol] + ")?timeout=3000 \
+-DACTIVEMQ_USER=" + activemq_username + " \
+-DACTIVEMQ_PASSWORD=" + activemq_password + " \
+-DFILE_STORE_URL=" + ats_proto() + "://" + sfs_host + ":" + ports_cfg.sfs.http ~ "/alfresco/api/-default-/private/sfs/versions/1/file" %}
+{% endif %}
+{% set tengine_environment_temp = tengine_environment['JAVA_OPTS'] %}
+{% set java_opts = java_opts + " " + tengine_environment_temp|join(' ') %}
```
Comment (pmacius) [inline roles/transformers/templates/alfresco-tengine-aio.service.j2:39]: Im not sue if this is the best way to do it.

File: roles/transformers/templates/alfresco-tengine-aio.service.j2 [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+{%- macro im_cfg() -%}
+{%- if pkg_format == 'deb' -%}
+{{ imagemagick_home }}/lib/x86_64-linux-gnu/ImageMagick-{{ imagemagick_version.split('-')[0] -}}
+{%- else -%}
+{{ imagemagick_home }}
+{%- endif -%}
+{%- endmacro %}
+
+{%- macro standrad_output_macro() -%}
+{%- if pkg_format == 'deb' -%}
```
Comment (gionn) [inline roles/transformers/templates/alfresco-tengine-aio.service.j2:25]: probably better to use `ansible_os_family` or `ansible_distribution` if needs to be more specific

File: roles/transformers/templates/alfresco-tengine-aio.service.j2 [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+{%- macro im_cfg() -%}
+{%- if pkg_format == 'deb' -%}
+{{ imagemagick_home }}/lib/x86_64-linux-gnu/ImageMagick-{{ imagemagick_version.split('-')[0] -}}
+{%- else -%}
+{{ imagemagick_home }}
+{%- endif -%}
+{%- endmacro %}
+
+{%- macro standrad_output_macro() -%}
```
Comment (gionn) [inline roles/transformers/templates/alfresco-tengine-aio.service.j2:24]: also check typo

File: molecule/pki/verify.yml [hunk 4]
```diff
@@ -44,8 +44,39 @@
...
+      loop:
+        - file: alfresco-tengine-aio
+          pattern: GET Transform Config version
+          edition: Community
+        - file: alfresco-transform-router
+          pattern: GET Transform Config version
+          edition: Enterprise
+        - file: alfresco-shared-fs
+          pattern: TLS virtual host
+          edition: Enterprise
```
Comment (pmacius) [inline molecule/pki/verify.yml:76]: Decided to add this additional task due:
https://github.com/Alfresco/alfresco-ansible-deployment/actions/runs/9172660358/job/25219680867

File: molecule/pki/verify.yml [hunk 5]
```diff
@@ -44,8 +44,39 @@
...
+      when: ansible_os_family == 'RedHat'
+      ansible.builtin.shell:
+        cmd: journalctl -u {{ item }} > /var/log/alfresco/{{ item }}.log
+      loop:
+        - alfresco-tengine-aio
+        - alfresco-transform-router
+        - alfresco-shared-fs
+      become: true
+
+    - name: Check in logs a client did connect
```
Comment (gionn) [inline molecule/pki/verify.yml:57]: use set_facts to populate a var depending on the os family to be feed to the `loop` structure so you don't duplicate the entire slurp task

File: roles/transformers/tasks/main.yml [hunk 6]
```diff
@@ -108,36 +107,10 @@
...
-        group: "{{ group_name }}"
-        block: |
-          export LIBREOFFICE_HOME={{ libreoffice_home }}
-          export IMAGEMAGICK_HOME={{ imagemagick_home }}
-          export IMAGEMAGICK_DYN={{ im_lib }}
-          export IMAGEMAGICK_CONFIG={{ im_cfg }}/config-Q16HDRI
-          export IMAGEMAGICK_CODERS={{ im_cfg }}/modules-Q16HDRI/coders
-          export IMAGEMAGICK_EXE="{% if pkg_format == 'deb' %}${IMAGEMAGICK_HOME}/bin/{% endif %}convert"
-        insertafter: EOF
+    - name: Delete startup script
```
Comment (gionn) [inline roles/transformers/tasks/main.yml:110]:
```suggestion
    - name: Delete deprecated startup script
```

minor: move this task where the `Add AIO startup script` was

File: molecule/pki/verify.yml [hunk 7]
```diff
@@ -44,6 +44,29 @@
...
+        log_files_names:
+          - file: "{% if ansible_os_family == 'RedHat' %}alfresco-tengine-aio{% else %}ats-ate-aio{% endif %}"
+            pattern: GET Transform Config version
+            edition: Community
+          - file: "{% if ansible_os_family == 'RedHat' %}alfresco-transform-router{% else %}ats-atr{% endif %}"
+            pattern: GET Transform Config version
+            edition: Enterprise
+          - file: "{% if ansible_os_family == 'RedHat' %}alfresco-shared-fs{% else %}ats-shared-fs{% endif %}"
+            pattern: TLS virtual host
+            edition: Enterprise
```
Comment (gionn) [inline molecule/pki/verify.yml:68]: oh wait, but you can just make sure that both platforms produced the same logs file by using a more complex loop structure like:

```
- service_name: alfresco-tengine-aio
  file_name: ats-ate-aio
```

**Review by gionn — CHANGES_REQUESTED** [review]

**Review by gionn — APPROVED** [review]

**Pull request merged**